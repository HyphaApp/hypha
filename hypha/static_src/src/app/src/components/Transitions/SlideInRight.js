import React from 'react'
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';


const SlideInRight = ({ children, in: inProp }) => {
    const duration = 250;

    const defaultStyle = {
        transition: `transform ${duration}ms ease-in-out`,
        transform: 'translate3d(0, 0, 0)',
        position: 'absolute',
        zIndex: 2,
        width: '100%'
    }

    const transitionStyles = {
        entering: { transform: 'translate3d(0, 0, 0)' },
        entered: { transform: 'translate3d(100%, 0, 0)' },
        exiting: { transform: 'translate3d(100%, 0, 0)' },
        exited: { transform: 'translate3d(0, 0, 0)' }
    };
    return (
        <Transition in={inProp} timeout={duration}>
            {(state) => (
                <div style={{ ...defaultStyle, ...transitionStyles[state] }}>
                    {children}
                </div>
            )}
        </Transition>
    )
}

SlideInRight.propTypes = {
    children: PropTypes.node,
    in: PropTypes.bool,
}

export default SlideInRight
