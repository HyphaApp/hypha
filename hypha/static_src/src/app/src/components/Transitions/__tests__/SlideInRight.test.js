import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import SlideInRight from "../SlideInRight";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Slide in right component", () => {
        
    const children = <span>Children</span>
    const subject = mount(<SlideInRight>{children}</SlideInRight>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    test('render a Slide in right component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
