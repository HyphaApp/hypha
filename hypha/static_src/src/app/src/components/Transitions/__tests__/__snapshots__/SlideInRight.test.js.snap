// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Slide in right component render a Slide in right component 1`] = `
<SlideInRight>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={250}
    unmountOnExit={false}
  >
    <div
      style={
        Object {
          "position": "absolute",
          "transform": "translate3d(0, 0, 0)",
          "transition": "transform 250ms ease-in-out",
          "width": "100%",
          "zIndex": 2,
        }
      }
    >
      <span>
        Children
      </span>
    </div>
  </Transition>
</SlideInRight>
`;
